rust/fory-core/src/serializer/marker.rs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Serializer implementations for marker types like `PhantomData<T>`.
+
+use crate::error::Error;
+use crate::resolver::context::{ReadContext, WriteContext};
+use crate::resolver::type_resolver::TypeResolver;
+use crate::serializer::{ForyDefault, Serializer};
+use crate::types::TypeId;
+use std::marker::PhantomData;
+
+impl<T: 'static> Serializer for PhantomData<T> {
+    #[inline(always)]
+    fn fory_write_data(&self, _context: &mut WriteContext) -> Result<(), Error> {
+        // PhantomData has no data to write
+        Ok(())
+    }
+
+    #[inline(always)]
+    fn fory_read_data(_context: &mut ReadContext) -> Result<Self, Error> {
+        // PhantomData has no data to read
+        Ok(PhantomData)
+    }
+
+    #[inline(always)]
+    fn fory_reserved_space() -> usize {
+        0
+    }
+
+    #[inline(always)]
+    fn fory_get_type_id(_: &TypeResolver) -> Result<u32, Error> {
+        // Use UNKNOWN to skip type registry lookup - PhantomData<T> has no runtime data
+        Ok(TypeId::UNKNOWN as u32)
```

Comment:
We have a `NONE` type id, could we use that instead for `PhantomData` and `Unit`?

Reply:
Seems we don't have any `None` in the type ids.

Reply:
I added for C++, but have not added to rust:

cpp/fory/type/type.h
```c++
  UNION = 38,
  // a null value with no data.
  NONE = 39,
```

Reply:
for struct, when we skip a field of type UNKNOWN, we will keep read more data for this field. But if we usign NONE, we can update skip.rs to return early

Reply:
Make sense, I think we can merge this PR first, in next PR, I'll add `None` and adjust the `PhantomData` and `()` to `None`.